src/coreclr/jit/emitloongarch64.cpp

```diff
@@ -0,0 +1,6431 @@
+                emitGCvarDeadUpd(adr + ofs, dst2 DEBUG_ARG(varNum));
+        }
+        // if (emitInsWritesToLclVarStackLocPair(id))
+        //{
+        //    unsigned ofs2 = ofs + TARGET_POINTER_SIZE;
+        //    if (id->idGCrefReg2() != GCT_NONE)
+        //    {
+        //        emitGCvarLiveUpd(adr + ofs2, varNum, id->idGCrefReg2(), *dp);
+        //    }
+        //    else
+        //    {
+        //        // If the type of the local is a gc ref type, update the liveness.
+        //        var_types vt;
+        //        if (varNum >= 0)
+        //        {
+        //            // "Regular" (non-spill-temp) local.
+        //            vt = var_types(emitComp->lvaTable[varNum].lvType);
+        //        }
+        //        else
+        //        {
+        //            TempDsc* tmpDsc = codeGen->regSet.tmpFindNum(varNum);
+        //            vt              = tmpDsc->tdTempType();
+        //        }
+        //        if (vt == TYP_REF || vt == TYP_BYREF)
+        //            emitGCvarDeadUpd(adr + ofs2, *dp);
+        //    }
+        //}
+    }
+
+#ifdef DEBUG
+    /* Make sure we set the instruction descriptor size correctly */
+
+    // size_t expected = emitSizeOfInsDsc(id);
+    // assert(sz == expected);
+
+    if (emitComp->opts.disAsm || emitComp->verbose)
+    {
+        code_t* cp = (code_t*)*dp;
+        while ((BYTE*)cp != dst)
+        {
+            emitDisInsName(*cp, (BYTE*)cp, id);
+            cp++;
+        }
+        // emitDispIns(id, false, dspOffs, true, emitCurCodeOffs(odst), *dp, (dst - *dp), ig);
+    }
+
+    if (emitComp->compDebugBreak)
+    {
+        // For example, set JitBreakEmitOutputInstr=a6 will break when this method is called for
+        // emitting instruction a6, (i.e. IN00a6 in jitdump).
+        if ((unsigned)JitConfig.JitBreakEmitOutputInstr() == id->idDebugOnlyInfo()->idNum)
+        {
+            assert(!"JitBreakEmitOutputInstr reached");
+        }
+    }
+#endif
+
+    /* All instructions are expected to generate code */
+
+    assert(*dp != dst);
+
+    *dp = dst;
+
+    return sz;
+}
+
+/*****************************************************************************/
+/*****************************************************************************/
+
+#ifdef DEBUG
+
+/****************************************************************************
+ *
+ *  Display the given instruction.
+ */
+
+// NOTE: At least 32bytes within dst.
+void emitter::emitDisInsName(code_t code, const BYTE* dst, instrDesc* id)
+{
+    const BYTE* insstrs = dst;
+
+    if (!code)
+    {
+        printf("LOONGARCH invalid instruction: 0x%x\n", code);
+        assert(!"invalid inscode on LOONGARCH!");
+        return;
+    }
+
+    // clang-format off
+    const char * const regName[] = {"zero", "ra", "tp", "sp", "a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7", "t0", "t1", "t2", "t3", "t4", "t5", "t6", "t7", "t8", "x0", "fp", "s0", "s1", "s2", "s3", "s4", "s5", "s6", "s7", "s8"};
```

Comment:
Isn't there already a array of register names? `getRegName()`

Reply:
Yes you are right.

Could I keep this for LA ?
I defined it here not only for the debug mode, this function is nearly independed others which can be quickly porting to the release mode when debuging needed.

Thanks

Reply:
OK, thanks very much.
I will replace `regName` with `RegNames` which is not difficult for release mode.